### PR TITLE
AK-16248 Implement update for connector aws_vpc, azure_vnet and ipsec.

### PR DESCRIPTION
**Pull request by spikewang**
Implement update function for connector aws_vpc, azure_vnet and ipsec.

### CODE REVIEW

**Comment by spikewang** [comment]
I have manually tested several cases and still thinking about how to document the limitation properly...

File: alkira/resource_alkira_connector_ipsec.go [hunk 1]
```diff
@@ -345,6 +345,66 @@ func resourceConnectorIPSecRead(d *schema.ResourceData, m interface{}) error {
...
+		Group:          d.Get("group").(string),
+		Name:           d.Get("name").(string),
+		PolicyOptions:  policyOptions,
+		RoutingOptions: routingOptions,
+		SegmentOptions: segmentOptions,
+		Segments:       segments,
+		Sites:          sites,
+		Size:           d.Get("size").(string),
+		VpnMode:        vpnMode,
+	}
```
Comment (rvichery) [inline alkira/resource_alkira_connector_ipsec.go:399]: Seems to be the same as L278 to L329, should we consider a `prepareConnectorIPSecRequest` function so we don't duplicate that logic

File: alkira/resource_alkira_connector_aws_vpc.go [hunk 2]
```diff
@@ -181,7 +181,49 @@ func resourceConnectorAwsVpcRead(d *schema.ResourceData, m interface{}) error {
...
+		CustomerName:   client.Username,
+		CustomerRegion: d.Get("aws_region").(string),
+		Group:          d.Get("group").(string),
+		Name:           d.Get("name").(string),
+		Segments:       segments,
+		Size:           d.Get("size").(string),
+		VpcId:          d.Get("vpc_id").(string),
+		VpcOwnerId:     d.Get("aws_account_id").(string),
+		VpcRouting:     vpcRouting,
+	}
```
Comment (rvichery) [inline alkira/resource_alkira_connector_aws_vpc.go:221]: Same comment as for IPSec, we should probably consider having a single function which prepare the request based on the ResourceData

File: alkira/resource_alkira_connector_aws_vpc.go [hunk 3]
```diff
@@ -181,7 +181,49 @@ func resourceConnectorAwsVpcRead(d *schema.ResourceData, m interface{}) error {
...
+
+	log.Printf("[INFO] Updateing Connector (AWS-VPC) %s", d.Id())
```
Comment (rvichery) [inline alkira/resource_alkira_connector_aws_vpc.go:223]:
```suggestion
	log.Printf("[INFO] Updating Connector (AWS-VPC) %s", d.Id())
```

**Review by rvichery — COMMENTED** [review]
Nothing wrong with the changes, but I feel we could simplify by having helpers create these complex structures (and prevent code duplication).

File: alkira/resource_alkira_connector_ipsec.go (hunk 1 above)
Comment (spikewang) [reply alkira/resource_alkira_connector_ipsec.go:399]: Yep, good idea...

File: alkira/resource_alkira_connector_aws_vpc.go (hunk 2 above)
Comment (spikewang) [reply alkira/resource_alkira_connector_aws_vpc.go]: Done

**Review by bchakrav1 — APPROVED** [review]
Looks good.

**Review by rvichery — APPROVED** [review]
Looks neat! Thanks!

**Pull request merged**